### PR TITLE
Enhancements for Classification

### PR DESCRIPTION
**Pull request by ss8319**
1) Added Weighted CE Loss by default

To do:
- Add the train/val/test fold class balance/distribution visualisation
- Think more abt stratification and keeping 'meaningful' class balance across train/val/test

### CODE REVIEW

**Comment by ss8319** [comment]
Added a new feature that allows us to visualise the train/val/test fold class balance/distribution across different folds

`training_params:
  visualize_fold_distribution: true`

Example Output for POS
Overall Dataset Distribution:
  Total samples: 5030
  Class 0: 1221 (24.3%)
  Class 1: 953 (18.9%)
  Class 2: 290 (5.8%)
  Class 3: 286 (5.7%)
  Class 4: 2280 (45.3%)

Per-Fold Distribution:
--------------------------------------------------------------------------------
Fold 1:
  Train (3018 samples): Class 0: 741 (24.6%) | Class 1: 576 (19.1%) | Class 2: 180 (6.0%) | Class 3: 155 (5.1%) | Class 4: 1366 (45.3%)      
  Val   (1006 samples): Class 0: 256 (25.4%) | Class 1: 184 (18.3%) | Class 2: 47 (4.7%) | Class 3: 66 (6.6%) | Class 4: 453 (45.0%)
  Test  (1006 samples): Class 0: 224 (22.3%) | Class 1: 193 (19.2%) | Class 2: 63 (6.3%) | Class 3: 65 (6.5%) | Class 4: 461 (45.8%)

**Comment by ss8319** [comment]
@itamarJa Perhaps you can throw this run on the GPU cluster. With the Weighted CE implementation. It does take awhile for me to run on my laptop. You can also set `early_stopping_metric: f1` instead of AUCROC. 
I will target this branch `dev/enhancements` for other potential enhancements for classification task

**Comment by ss8319** [comment]
You can run it as it is. Already has the weighted CE Loss in this branch

File: metrics/classification_metrics.py [hunk 1]
```diff
@@ -63,12 +63,48 @@ def bce_metric(predicted: torch.Tensor, groundtruth: torch.Tensor) -> float:
...
+        num_classes = predicted.shape[-1]
+        weight_tensor = torch.ones(num_classes, dtype=predicted.dtype, device=predicted.device)
+        
+        for class_idx, weight in zip(unique_classes, class_weights):
+            if 0 <= class_idx < num_classes:
+                weight_tensor[class_idx] = weight
+        
+        return F.cross_entropy(predicted, groundtruth.long(), weight=weight_tensor)
+        
+    except Exception as e:
```
Comment (zparis99) [inline metrics/classification_metrics.py]: I'm hesitant with a catch-all here. I'd rather just have it fail if the weight function breaks

**Review by zparis99 — COMMENTED** [review]
Looks good! Just wanted to make sure we maintain the old cross entropy as well.
